### DISCUSSION TITLE
integrations/paddlepaddle/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17657] [Docs]
# integrations/paddlepaddle/

Learn how to export YOLO11 models to PaddlePaddle format for enhanced performance, flexibility, and deployment across various platforms and devices.

https://docs.ultralytics.com/integrations/paddlepaddle/

**Comment by cadpm** [comment]
Dear team, your converter does not work with PaddleX as it does not generate the YAML file in the inference folder, making it impossible to deploy the converted model. PaddlePaddle cannot teach how to work with this file due to licensing conflicts.

n fact, nowhere in the PaddlePaddle or PaddleX repositories do they teach how to work with external or converted models.

**Reply by glenn-jocher** [reply]
@cadpm thank you for bringing this to our attention. While the Ultralytics export tool facilitates converting YOLO11 models to PaddlePaddle format, it currently doesn't generate a YAML file for PaddleX inference. For specific guidance on handling PaddlePaddle models in PaddleX, consulting their official documentation or community forums might provide additional insights. If you encounter issues with the export process itself, please refer to our [Ultralytics documentation](https://docs.ultralytics.com/integrations/paddlepaddle/) for troubleshooting tips.

**Reply by cadpm** [reply]
Unfortunately, I haven't been able to find a solution on either side. For this reason, I have written to you to report this issue. If I ask in the Paddle forum, no one has worked on this issue, and the same happens on the YOLOv11 side—they don't work with Paddle.

**Reply by glenn-jocher** [reply]
@cadpm thank you for reporting this issue. While our export tool supports conversion to PaddlePaddle, integration with PaddleX requires additional steps beyond our current documentation. I recommend reaching out to the PaddlePaddle community for specific guidance on importing and using converted models with PaddleX, as they might have insights on handling this integration. If you need further assistance with the export process itself, feel free to check our [PaddlePaddle export guide](https://docs.ultralytics.com/integrations/paddlepaddle/).

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics and exploring our integrations! 🚀 We suggest visiting the [Docs](https://docs.ultralytics.com) for users looking to delve into various integration possibilities. You can explore our detailed guide on exporting YOLO models, such as your interest in PaddlePaddle [here](https://docs.ultralytics.com/integrations/paddlepaddle/).

If this is a 🐛 Bug Report or you are facing any issues, we would appreciate it if you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist in debugging.

In case you have a ❓ Question regarding custom operations or integrations, please supply detailed information regarding your use case. Sharing logs, code snippets, and screenshots enhances support experience and efficiency.

Join our vibrant community for more in-depth interactions or real-time chats. Head to [Discord](https://ultralytics.com/discord) 🎧 for quick consultations, engage in comprehensive discussions on [Discourse](https://community.ultralytics.com), or explore threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are running the latest version of `ultralytics` including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) setting with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to confirm that your issue is not already addressed:

```bash
pip install -U ultralytics
```

## Environments

YOLO functions efficiently in these fully configured environments (with essential dependencies such as [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) pre-installed):

- **Notebooks** with free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** with Deep Learning VM. Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) ensuring YOLO operations across macOS, Windows, and Ubuntu are functioning as expected, is passing its tests. An Ultralytics engineer will respond to provide further assistance soon!
